src/components/Contacts/ContactDetails/ContactDetailsTab/Other/EditContactOtherModal/EditContactOtherModal.tsx

```diff
@@ -481,48 +484,107 @@ export const EditContactOtherModal: React.FC<EditContactOtherModalProps> = ({
-                <ContactInputWrapper>
-                  <Autocomplete
-                    loading={loadingChurchOptions}
-                    autoSelect
-                    autoHighlight
-                    freeSolo
-                    options={churches}
-                    renderInput={(params): ReactElement => (
+
+                  <Grid item xs={12} sm={6}>
+                    <ContactInputWrapper>
+                      <Autocomplete
+                        loading={loadingChurchOptions}
+                        autoSelect
+                        autoHighlight
+                        freeSolo
+                        options={churches}
+                        renderInput={(params): ReactElement => (
+                          <TextField
+                            {...params}
+                            label={t('Church')}
+                            InputProps={{
+                              ...params.InputProps,
+                              endAdornment: (
+                                <>
+                                  {loadingChurchOptions && (
+                                    <CircularProgress
+                                      color="primary"
+                                      size={20}
+                                    />
+                                  )}
+                                  {params.InputProps.endAdornment}
+                                </>
+                              ),
+                            }}
+                          />
+                        )}
+                        value={churchName || ''}
+                        onChange={(_, churchName): void => {
+                          setFieldValue('churchName', churchName);
+                        }}
+                      />
+                    </ContactInputWrapper>
+                  </Grid>
+                  <Grid item xs={12} sm={6}>
+                    <ContactInputWrapper>
                       <TextField
-                        {...params}
-                        label={t('Church')}
-                        InputProps={{
-                          ...params.InputProps,
-                          endAdornment: (
-                            <>
-                              {loadingChurchOptions && (
-                                <CircularProgress color="primary" size={20} />
-                              )}
-                              {params.InputProps.endAdornment}
-                            </>
-                          ),
+                        label={t('Website')}
+                        value={website}
+                        onChange={handleChange('website')}
+                        inputProps={{ 'aria-label': t('Website') }}
+                        fullWidth
+                      />
+                    </ContactInputWrapper>
+                  </Grid>
+                  <Grid item xs={12}>
+                    <ContactInputWrapper>
```

Comment:
Would you be able to use `src/components/Task/Modal/Form/Inputs/ContactsAutocomplete/ContactsAutocomplete.tsx` to reduce duplication? If so, we should probably move that input out of `src/components/Task` and into `src/components/Shared`.

Reply:
That makes sense. It sounds like it's a good idea to keep it how you have it now.